internal/metrics/workload.go

```diff
@@ -0,0 +1,109 @@
+package metrics
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"git.sr.ht/~spc/go-log"
+
+	"github.com/jakub-dzon/k4e-device-worker/internal/workload/podman"
+	"github.com/jakub-dzon/k4e-operator/models"
+)
+
+const (
+	defaultInterval = 60
+)
+
+type WorkloadMetrics struct {
+	latestConfig   models.WorkloadList
+	daemon         MetricsDaemon
+	workloadConfig map[string]*models.Workload
+	lock           sync.RWMutex
+}
+
+func NewWorkloadMetrics(daemon MetricsDaemon) *WorkloadMetrics {
+	return &WorkloadMetrics{daemon: daemon}
+}
+
+func (wrkM *WorkloadMetrics) getWorkload(workloadName string) *models.Workload {
+	wrkM.lock.Lock()
+	defer wrkM.lock.Unlock()
+	return wrkM.workloadConfig[workloadName]
+}
+
+func (wrkM *WorkloadMetrics) Update(config models.DeviceConfigurationMessage) error {
```

Comment:
Why do we need to return `error`? It seems we always return `nil`

Reply:
It's part of the interface for `Observe()`